conformance/utils/roundtripper/roundtripper.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2022 The Kubernetes Authors.
```

Comment:
I'd recommend starting with a single `conformance/utils/http` package which includes this file.
Any reason to have so manage packages upfront?

Reply:
Since this is an interface that implementations can override, I've wanted to keep it as generic and possible and leave the door open for at least TCP. That's a ways out though, and I'm not really sure how we'd structure that. The idea was to separate the interface + default implementation from HTTP specific helper funcs.